samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -143,14 +171,18 @@ public Startpoint readStartpoint(SystemStreamPartition ssp, TaskName taskName) {
     Preconditions.checkState(!stopped, "Underlying metadata store not available");
     Preconditions.checkNotNull(ssp, "SystemStreamPartition cannot be null");
 
-    byte[] startpointBytes = metadataStore.get(toStoreKey(ssp, taskName));
+    byte[] startpointBytes = readWriteStore.get(toReadWriteStoreKey(ssp, taskName));
 
-    if (Objects.nonNull(startpointBytes)) {
-      Startpoint startpoint = startpointSerde.fromBytes(startpointBytes);
-      if (Instant.now().minus(DEFAULT_EXPIRATION_DURATION).isBefore(Instant.ofEpochMilli(startpoint.getCreationTimestamp()))) {
-        return startpoint; // return if deserializable and if not stale
+    if (ArrayUtils.isNotEmpty(startpointBytes)) {
+      try {
+        Startpoint startpoint = objectMapper.readValue(startpointBytes, Startpoint.class);
+        if (Instant.now().minus(DEFAULT_EXPIRATION_DURATION).isBefore(Instant.ofEpochMilli(startpoint.getCreationTimestamp()))) {
+          return startpoint; // return if deserializable and if not stale
+        }
+        LOG.warn("Stale Startpoint: {} was read. Ignoring.", startpoint);
```

Comment:
nit: Minor. Would be better to include the expiration-duration and startpoint-timestamp in the log-message. 

```suggestion
        LOG.warn("Creation timestamp: {} of startpoint: {} has crossed the expiration duration: {}. Ignoring it", startpoint.getTimestamp(), startpoint, DEFAULT_EXPIRATION_DURATION);
```

Reply:
Will do